[HUD] Display on hud table the autorevert decisions

jeanschmidt opened the pull request:
* Highlight hud autorevert decisions on main hud table with a different background
* Add a banner on hover for autoreverted commits on main hud table
* Special background for signals that are flagged to autorevert, making them easier to identify
* Adds a banner on commit page about the autorevert and the identified signals

jeanschmidt commented:
<img width="1197" height="442" alt="Screenshot 2025-10-06 at 18 48 30" src="https://github.com/user-attachments/assets/0932f28f-5594-4690-8753-eae3a5ac373e" />
<img width="1197" height="442" alt="Screenshot 2025-10-06 at 18 48 12" src="https://github.com/user-attachments/assets/72fe71d5-00c8-4ab4-a22f-7e5928e4f417" />

izaitsevfb commented on torchci/lib/utilization/fetchUtilization.test.ts:
```diff
@@ -213,7 +213,7 @@ describe("Test flattenTS to flatten timestamp", () => {
     // assert log
     expect(logSpy).toHaveBeenCalledWith(
-      `Warning: Error parsing JSON:SyntaxError: Expected property name or '}' in JSON at position 1 for data string '{{}dsad}'`
+      `Warning: Error parsing JSON:SyntaxError: Expected property name or '}' in JSON at position 1 (line 1 column 2) for data string '{{}dsad}'`
```
accedental change?

jeanschmidt replied:
no tried to fix, but the message is not correct, I can remove the change

izaitsevfb commented:
Several notes:

0. Please address accidental changes.
1. The job marking logic doesn't work when grouping is enabled:

<img width="101" height="70" alt="image" src="https://github.com/user-attachments/assets/9f6a0330-4dc8-4554-a6e7-9cc0ba416ae6" />
<img width="1784" height="260" alt="image" src="https://github.com/user-attachments/assets/80e4dfc5-b004-45a9-b033-4b2ec359718b" />
please extract the common logic that determines if job should be marked and reuse it for both codepaths, also check other settings, like failures monsterization.

2. the red border is very similar to "blocks viable strict". maybe let's choose another color to avoid confusion?

<img width="56" height="42" alt="image" src="https://github.com/user-attachments/assets/b40456ef-7f79-469b-acab-e447d15e685b" />

jeanschmidt commented:
@izaitsevfb 

> Please address accidental changes.

Note it wasn't a accidental changes, just gave up on fixing the test for the newer version of the library. End up updating anyway.

> The job marking logic doesn't work when grouping is enabled:
> image
> image
> please extract the common logic that determines if job should be marked and reuse it for both codepaths, also check other settings, like failures monsterization.

Yeah, ok, good point, extracted the logic and reused for both the grouped and expanded views, I wasn't going to support the expanded views, but I guess this is a good point that support it.

> the red border is very similar to "blocks viable strict". maybe let's choose another color to avoid confusion?

changed to orange

izaitsevfb commented on torchci/components/job/GroupJobConclusion.tsx:
```diff
@@ -430,26 +489,42 @@ function ToolTip({
+            style={{
+              display: "block",
+              fontWeight: isAutorevert ? "bold" : "normal",
+              color: isAutorevert ? "#d73a49" : undefined,
```
yeah, I think it makes sense to make it a css class, because currently the style is duplicating in two places

izaitsevfb commented on torchci/components/commit/AutorevertBanner.tsx:
```diff
@@ -0,0 +1,133 @@
+  // Handle case where arrays might be undefined or have different structure
+  const workflows = autorevertData.workflows || [];
+  const sourceSignalKeys = autorevertData.source_signal_keys || [];
+  const jobIds = autorevertData.job_ids || [];
```
those are unused!

jeanschmidt replied:
good point!

jeanschmidt commented:
Abandoning this PR, 

I'll redo this change with a bit of more time by next week or further.
